### PR TITLE
TFKUBE-461: Update license in shared home if restoring from snapshot

### PR DESCRIPTION
**Pull request by nanux**
## Pull request description

The license key is stored in the EBS volume with the shared home content when we are restoring from a snapshot. The license wasn't updated with the one used in the `config.tfvars`.

The change adds a command to replace the stored license with the value defined in the config file.

## Checklist
- [ ] I have successful end to end tests run (with & without domain)
- [x] I have added unit tests (if applicable)
- [x] I have user documentation (if applicable)

### CODE REVIEW

File: modules/products/confluence/locals.tf [hunk 1]
```diff
@@ -87,4 +87,10 @@ locals {
       ]
     }
   }) : yamlencode({})
+
+  # updates base url (in case we are restoring from snapshot)
+  cmd_psql_update = "BASE_URL_TO_REPLACE=$(PGPASSWORD=${var.db_master_password} psql postgresql://${module.database.rds_endpoint}/${local.product_name} -U ${var.db_master_username} -Atc \"select BANDANAVALUE from BANDANA where BANDANACONTEXT = '_GLOBAL' and BANDANAKEY = 'atlassian.confluence.settings';\" | grep -i '<baseurl>'); PGPASSWORD=${var.db_master_password} psql postgresql://${module.database.rds_endpoint}/${local.product_name} -U ${var.db_master_username} -c \"update BANDANA set BANDANAVALUE = replace(BANDANAVALUE, '$${BASE_URL_TO_REPLACE}', '<baseUrl>${local.confluence_ingress_url}</baseUrl>') where BANDANACONTEXT = '_GLOBAL' and BANDANAKEY = 'atlassian.confluence.settings';\""
+
+  # updates license in shared home (in case we are restoring from snapshot)
+  cmd_license_update = "sed -i 's|<property name=\"atlassian.license.message\">.*</property>|<property name=\"atlassian.license.message\">${var.confluence_configuration["license"]}</property>|g' /shared-home/confluence.cfg.xml"
```
Comment (yzha645) [inline modules/products/confluence/locals.tf:95]: I had a brief look at it a couple of days ago, and was thinking it's too much trouble to escape `/` in the license value. `|` is a better option. Good to know!

File: modules/products/confluence/pre_install.tf [hunk 2]
```diff
@@ -12,9 +12,19 @@ resource "kubernetes_job" "pre_install" {
...
+          volume_mount {
+            mount_path = "/shared-home"
+            name       = "nfs-shared-home"
+          }
         }
         restart_policy = "Never"
+        volume {
+          name = "nfs-shared-home"
+          persistent_volume_claim {
+            claim_name = module.nfs.nfs_claim_name
```
Comment (yzha645) [inline modules/products/confluence/pre_install.tf:25]: Do we need to depends_on module.nfs as well?

Comment (yzha645) [inline modules/products/confluence/pre_install.tf:25]: Do we need to depends_on module.nfs as well?

Comment (nanux) [reply modules/products/confluence/pre_install.tf:25]: I think the fact that it defines the variable from the module will inform Terraform that there is a dependency and it doesn't need to have `depends_on`

**Review by yzha645 — APPROVED** [review]

**Pull request merged**